### PR TITLE
GEOS-9290 jdbcconfig: update catalog info references

### PR DESCRIPTION
**Pull request by NielsCharlier**

### CODE REVIEW

File: src/community/jdbcconfig/src/main/java/org/geoserver/jdbcconfig/internal/ConfigDatabase.java [hunk 1]
```diff
@@ -1678,4 +1692,110 @@ public void handleSettingsAdded(SettingsInfo settings) {
...
+        public void visit(CoverageInfo coverage) {
+            if (coverage.getNamespace() != null) {
+                coverage.setNamespace(
+                        getById(coverage.getNamespace().getId(), NamespaceInfo.class));
+            }
+            coverage.setStore(getById(coverage.getStore().getId(), StoreInfo.class));
+        }
+
+        @Override
+        public void visit(LayerInfo layer) {
```
Comment (sikeoka) [inline src/community/jdbcconfig/src/main/java/org/geoserver/jdbcconfig/internal/ConfigDatabase.java]: Does the Resource also need to be updated?

File: src/community/jdbcconfig/src/main/java/org/geoserver/jdbcconfig/internal/ConfigDatabase.java [hunk 2]
```diff
@@ -1678,4 +1692,110 @@ public void handleSettingsAdded(SettingsInfo settings) {
...
+
+        @Override
+        public void visit(StyleInfo style) {
+            if (style.getWorkspace() != null) {
+                style.setWorkspace(getById(style.getWorkspace().getId(), WorkspaceInfo.class));
+            }
+        }
+
+        @Override
+        public void visit(LayerGroupInfo layerGroup) {
```
Comment (sikeoka) [inline src/community/jdbcconfig/src/main/java/org/geoserver/jdbcconfig/internal/ConfigDatabase.java]: Do the RootLayer, RootLayerStyle and Styles also need to be updated?

File: src/community/jdbcconfig/src/main/java/org/geoserver/jdbcconfig/internal/ConfigDatabase.java [hunk 3]
```diff
@@ -1678,4 +1692,110 @@ public void handleSettingsAdded(SettingsInfo settings) {
...
+
+        @Override
+        public void visit(WMSLayerInfo wmsLayer) {
+            if (wmsLayer.getNamespace() != null) {
+                wmsLayer.setNamespace(
+                        getById(wmsLayer.getNamespace().getId(), NamespaceInfo.class));
+            }
+            wmsLayer.setStore(getById(wmsLayer.getStore().getId(), StoreInfo.class));
+        }
+    }
```
Comment (sikeoka) [inline src/community/jdbcconfig/src/main/java/org/geoserver/jdbcconfig/internal/ConfigDatabase.java]: It probably wouldn't hurt to override the default implementations of visit(WMTSStoreInfo store) and visit(WMTSLayerInfo layer).

File: src/community/jdbcconfig/src/main/java/org/geoserver/jdbcconfig/internal/ConfigDatabase.java (hunk 1 above)
Comment (NielsCharlier) [reply src/community/jdbcconfig/src/main/java/org/geoserver/jdbcconfig/internal/ConfigDatabase.java]: no, there is a separate mechanism for that, the layer gets invalidated if the resource changes

File: src/community/jdbcconfig/src/main/java/org/geoserver/jdbcconfig/internal/ConfigDatabase.java (hunk 2 above)
Comment (NielsCharlier) [reply src/community/jdbcconfig/src/main/java/org/geoserver/jdbcconfig/internal/ConfigDatabase.java]: indeed

File: src/community/jdbcconfig/src/main/java/org/geoserver/jdbcconfig/internal/ConfigDatabase.java (hunk 3 above)
Comment (NielsCharlier) [reply src/community/jdbcconfig/src/main/java/org/geoserver/jdbcconfig/internal/ConfigDatabase.java]: why, they don't seem to have any references?

File: src/community/jdbcconfig/src/main/java/org/geoserver/jdbcconfig/internal/ConfigDatabase.java (hunk 2 above)
Comment (NielsCharlier) [reply src/community/jdbcconfig/src/main/java/org/geoserver/jdbcconfig/internal/ConfigDatabase.java]: commit added for this

File: src/community/jdbcconfig/src/main/java/org/geoserver/jdbcconfig/internal/ConfigDatabase.java (hunk 3 above)
Comment (sikeoka) [reply src/community/jdbcconfig/src/main/java/org/geoserver/jdbcconfig/internal/ConfigDatabase.java]: Those two methods were added using Java 8 default methods.

File: src/community/jdbcconfig/src/main/java/org/geoserver/jdbcconfig/internal/ConfigDatabase.java (hunk 2 above)
Comment (sikeoka) [reply src/community/jdbcconfig/src/main/java/org/geoserver/jdbcconfig/internal/ConfigDatabase.java]: LayerGroupInfo styles are a List (not a Set like LayerInfo) and the order of the styles does matter.

File: src/community/jdbcconfig/src/main/java/org/geoserver/jdbcconfig/internal/ConfigDatabase.java (hunk 3 above)
Comment (NielsCharlier) [reply src/community/jdbcconfig/src/main/java/org/geoserver/jdbcconfig/internal/ConfigDatabase.java]: I understand, but I see no reason to override them?

File: src/community/jdbcconfig/src/main/java/org/geoserver/jdbcconfig/internal/ConfigDatabase.java (hunk 2 above)
Comment (NielsCharlier) [reply src/community/jdbcconfig/src/main/java/org/geoserver/jdbcconfig/internal/ConfigDatabase.java]: Yes, indeed. Fixed this.

File: src/community/jdbcconfig/src/main/java/org/geoserver/jdbcconfig/internal/ConfigDatabase.java (hunk 3 above)
Comment (sikeoka) [reply src/community/jdbcconfig/src/main/java/org/geoserver/jdbcconfig/internal/ConfigDatabase.java]: Why would you need to do something for WMSStoreInfo and WMSLayerInfo but not WMTSStoreInfo and WMTSLayerInfo?

Comment (NielsCharlier) [reply src/community/jdbcconfig/src/main/java/org/geoserver/jdbcconfig/internal/ConfigDatabase.java]: sorry, you are right, it does ahve references! I changed it

**Comment by sikeoka** [comment]
Any chance you could make the PR a single commit instead of four commits?  Other than that, it looks good to me.

**Comment by NielsCharlier** [comment]
Thanks @sikeoka. you can squash merge with github :) Just fixed a style error and waiting for new build

**Pull request merged**